src/modalities/optimizers/optimizer_factory.py

```diff
@@ -84,7 +89,12 @@ def _assert_existence_of_weight_decay_groups_excluded(model: FSDP, weight_decay_
         }
         weight_decay_groups_excluded = ["embedding", "layernorm"]
     """
-    nn_model: NNModel = model.module
+    # FSDP 1
```

Comment:
We could encapsulate the logic in a helper function since it is used at least twice.